console-frontend/src/components/spy.rs

```diff
@@ -100,7 +109,13 @@ impl Component for Spy {
             Msg::Event(event) => {
                 // log::debug!("Pushing event: {:?}", event);
                 self.total_received += 1;
-                self.events.insert(0, Entry(*event));
+                let entry = Entry(*event);
+                if (self.device == entry.device() || self.device.is_empty())
```

Comment:
I am wondering if it would make sense reversing this check: check first if it is empty, then check for the equality. As checking for `is_empty` should be faster.

Then again, I am not sure Rust evaluates from left to right? Maybe worth checking.

Reply:
Good catch ! I did some googling, and it's evaluated left to right : https://doc.rust-lang.org/reference/expressions.html#expression-precedence